### DISCUSSION TITLE
What metadata is Navidrome using to read track ratings?

### DISCUSSION THREAD

**Discussion by ClementNerma** [navidrome/navidrome#3086] [Q&A] [answered]
Hi there!

I want to import a pretty big music library (~ 15k tracks) into Navidrome. But I absolutely want to keep ratings, which have been set using another application, and they unfortunately don't show up after importing in Navidrome's UI.

So my question is: what does Navidrome use to read the ratings from a given track? ID3 rating? If so, which format is used to decode it?

I can easily make a small program that rewrites the rating using Navidrome's own track rating format, but I first need to know what it's using.

Thanks in advance for your help!

**Comment by dweymouth** [comment, marked as answer]
Navidrome is multi-user and each user can have their own ratings so it doesn't use the ID3 rating tag at all. Ratings are completely internal to Navidrome's database. It wouldn't be too difficult though to write a script to update the ratings using the Subsonic API after Navidrome imports your library.

**Reply by ClementNerma** [reply]
I see, thanks for the answer :)
